docs/programming-oak.md

```diff
@@ -368,20 +368,47 @@ using the following script:
 ./scripts/push_example -e hello_world
 ```
 
+### Creating a Permissions File
+
+A permissions file is a `.toml` file provided by the host owner, as an
+additional layer of defense. This file specifies the features of Oak that are
+permitted for the applications running on the host. In particular, using this
+file, the host owner can enable or disable gRPC or HTTP connections. In
+addition, it is possible to specify an allowlist of external gRPC or HTTP
+services that the applications can connect to. Interaction with all other
+external servers will be prohibited. Connections to insecure HTTP servers will
+be allowed regardless of the allowlist! However, Oak's IFC guarantees that only
```

Comment:
I added this note here, but I think the permissions file should allow blocking insecure HTTP connections altogether. So, I think we want the Runtime to block insecure HTTP connections by default. But if a host owner wants it to be allowed, then a special token, e.g., `HTTP`, should be included in `allow_http_egress_addresses`. WDYT?